junit-platform-suite-commons/src/main/java/org/junit/platform/suite/commons/SuiteLauncherDiscoveryRequestBuilder.java

```diff
@@ -201,6 +207,16 @@ public LauncherDiscoveryRequest build() {
 		return delegate.build();
 	}
 
+	private List<DiscoverySelector> selectMethods(SelectMethod... methods) {
+		List<DiscoverySelector> selectors = new ArrayList<>();
+
+		for (SelectMethod selectMethod : methods) {
+			selectors.add(DiscoverySelectors.selectMethod(selectMethod.returnType().getName(), selectMethod.name(),
+				selectMethod.parameterTypes()));
+		}
+		return selectors;
```

Comment:
Consider using the streaming API to clean up this mapping operation.

The returned list should also not contain duplicates. The `.distinct()` can help with this.